src/index.js

```diff
@@ -142,149 +137,79 @@ class Node {
     this.swarm.close(callback)
   }
 
-  //
-  // Ping
-  //
-
-  // TODO
-  pingById (id, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
-    callback(new Error('not implemented yet'))
+  isOn () {
+    return this.isOnline
   }
 
-  // TODO
-  pingByMultiaddr (maddr, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
-    callback(new Error('not implemented yet'))
-  }
-
-  pingByPeerInfo (peerInfo, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
+  ping (peer, callback) {
+    assert(this.isOn(), OFFLINE_ERROR_MESSAGE)
+    const peerInfo = this._getPeerInfo(peer)
     callback(null, new Ping(this.swarm, peerInfo))
   }
 
-  //
-  // Dialing methods
-  //
-
-  // TODO
-  dialById (id, protocol, callback) {
-    // NOTE: dialById only works if a previous dial was made. This will
-    // change once we have PeerRouting
-
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
-
-    if (typeof protocol === 'function') {
-      callback = protocol
-      protocol = undefined
-    }
-
-    callback(new Error('not implemented yet'))
-  }
-
-  dialByMultiaddr (maddr, protocol, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
-
-    if (typeof protocol === 'function') {
-      callback = protocol
-      protocol = undefined
-    }
-
-    if (typeof maddr === 'string') {
-      maddr = multiaddr(maddr)
-    }
-
-    if (!mafmt.IPFS.matches(maddr.toString())) {
-      return callback(new Error('multiaddr not valid'))
-    }
-
-    const ipfsIdB58String = maddr.stringTuples().filter((tuple) => {
-      if (tuple[0] === IPFS_CODE) {
-        return true
-      }
-    })[0][1]
-
-    let peer
-    try {
-      peer = this.peerBook.getByB58String(ipfsIdB58String)
-    } catch (err) {
-      peer = new PeerInfo(PeerId.createFromB58String(ipfsIdB58String))
-    }
-
-    peer.multiaddr.add(maddr)
-    this.dialByPeerInfo(peer, protocol, callback)
-  }
-
-  dialByPeerInfo (peer, protocol, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
+  dial (peer, protocol, callback) {
+    assert(this.isOn(), OFFLINE_ERROR_MESSAGE)
+    const peerInfo = this._getPeerInfo(peer)
 
     if (typeof protocol === 'function') {
       callback = protocol
       protocol = undefined
     }
 
-    this.swarm.dial(peer, protocol, (err, conn) => {
+    this.swarm.dial(peerInfo, protocol, (err, conn) => {
       if (err) {
         return callback(err)
       }
-      this.peerBook.put(peer)
+      this.peerBook.put(peerInfo)
       callback(null, conn)
     })
   }
 
-  //
-  // Disconnecting (hangUp) methods
-  //
+  hangUp (peer, callback) {
+    assert(this.isOn(), OFFLINE_ERROR_MESSAGE)
+    const peerInfo = this._getPeerInfo(peer)
 
-  hangUpById (id, callback) {
-    // TODO
-    callback(new Error('not implemented yet'))
+    this.peerBook.removeByB58String(peerInfo.id.toB58String())
```

Comment:
I was trying to remember why this was added here and I just figured out. pre-dht era, I treated the PeerBook as our source of truth for peers we are connected, and so: https://github.com/ipfs/js-ipfs/blob/master/src/core/components/swarm.js#L12-L47 relies on that.

We could do a match with swarm.muxedConns, but a better solution is to actually have a little bit more state on the PeerInfo object I believe. Same goes for the address that gets picked, so that when we do log out the peers we are connected, we only print the addr we are actually using (like go-ipfs does).